File: Week4/Solution1/1.js
const maxDepth = (s) => {
  let count = 0;
  let maxCount = 0;
  const sLength = s.length;

  for (let i = 0; i < sLength; ++i) {
      if(s[i] === '(') ++count;
      else if(s[i] === ')') {
          if(maxCount < count) {
              maxCount = count;
          }
          --count;
      }
  }
  return maxCount;
};

File: Week4/Solution1/1.c
int maxDepth(char * s){
    int count = 0, maxCount = 0, i = 0;
    while(s[i] != '\0') {
        if(s[i] == '(') ++count;
        if(s[i] == ')') --count;
        if(maxCount < count) maxCount = count;
        ++i;
    }
    return maxCount;
}